P/P11Kit/build_tarballs.jl

```diff
@@ -3,26 +3,40 @@
 using BinaryBuilder, Pkg
 
 name = "P11Kit"
-version = v"0.24.1"
+version = v"0.25.5"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/p11-glue/p11-kit/releases/download/$(version)/p11-kit-$(version).tar.xz", "d8be783efd5cd4ae534cee4132338e3f40f182c3205d23b200094ec85faaaef8")
+    ArchiveSource("https://github.com/p11-glue/p11-kit/releases/download/$(version)/p11-kit-$(version).tar.xz",
+                  "04d0a86450cdb1be018f26af6699857171a188ac6d5b8c90786a60854e1198e5")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir
-meson --cross-file=${MESON_TARGET_TOOLCHAIN} --buildtype=release p11-kit-*
+cd ${WORKSPACE}/srcdir/p11-kit-*
 
-# Meson beautifully forces thin archives, without checking whether the dynamic linker
-# actually supports them: <https://github.com/mesonbuild/meson/issues/10823>.  Let's remove
-# the (deprecated...) `T` option to `ar`
-sed -i.bak 's/csrDT/csrD/' build.ninja
+if [[ ${target} = *-mingw32 ]]; then
 
-ninja -j${nproc}
-ninja install
-install_license p11-kit-*/COPYING
+    # `configure` does not work on Windows
+
+    meson setup --cross-file="${MESON_TARGET_TOOLCHAIN}" --buildtype=release builddir
+    meson compile -C builddir
+    meson install -C builddir
+
+else
+
+    # Meson is too clever to build for Apple. It produces command line
```

Comment:
I remember somewhat similar issues in the past, meson tries very often to be oversmart, failing badly. But sometimes there are also ways to work around it, have you considered that possibility or it's too complicated?

Reply:
I looked at it briefly but I don't know meson well enough to know where to start.

Do you have a pointer to another package with a work-around?

Reply:
This meson test fails, trying to detect `sizeof(unsigned long)`:
```
Running compile:
Working directory:  /workspace/srcdir/p11-kit-0.25.5/builddir/meson-private/tmpvtqqu7h4
Code:

        #include <stddef.h>
        int main(void) {
            unsigned long something;
            return 0;
        }
-----------
Command line: `/opt/bin/aarch64-apple-darwin20-libgfortran5-cxx11/aarch64-apple-darwin20-clang -I/workspace/destdir/include /workspace/srcdir/p11-kit-0.25.5/builddir/meson-private/tmpvtqqu7h4/testfile.c -o /workspace/srcdir/p11-kit-0.25.5/builddir/meson-private/tmpvtqqu7h4/output.obj -c -O0 -Werror=implicit-function-declaration -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument` -> 1
stderr:
clang: error: -Wl,-sdk_version,11.0: 'linker' input unused [-Werror,-Wunused-command-line-argument]
-----------
Checking for size of "unsigned long" : -1
```

Reply:
Why they use Werror..........

Reply:
I don't know. I couldn't figure out where it comes from. I think it's a meson thing.

I'm fixing it by patching our compiler wrappers.

Reply:
This comment in `BinaryBuilderBase` is wrong:
```
                # We also add `-Wno-unused-command-line-argument` so that if someone does something like
                # `clang -Werror -o foo a.o b.o`, it doesn't complain due to the fact that that is using
                # `clang` as a linker (and we have no real way to detect that in the wrapper), which will
                # cause `clang` to complain about compiler flags being passed in.
```
We can (and do) detect whether the wrapper is called as a linker.

Also, `min_macos_version_flags` always returns compile_only flags. It should return the `-Wl,...` flag as link-only flag. The line
```
            macos_version_flags = clang_use_lld ? (min_macos_version_flags()[1],) : min_macos_version_flags()
```
is also iffy; it should only check for `lld` for linker flags, not compiler flags.